File: ReSharper.FSharp/src/FSharp.TypeProvidersProtocol/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;
using JetBrains.Rider.Backend.Product;

namespace JetBrains.ReSharper.Plugins.FSharp.TypeProvidersProtocol
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<IRiderProductEnvironmentZone>
  {
  }
}

File: ReSharper.FSharp/src/FSharp.Common/src/Settings/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Settings

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.ReSharper.Plugins.FSharp
open JetBrains.Rider.Model

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<ILanguageFSharpZone>
    interface IRequire<IRiderModelZone>
